src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+PlausibeDeniabilityTests.swift

```diff
@@ -105,11 +126,37 @@ class HTTPClientPlausibleDeniabilityTests: CWATestCase {
 		let client = HTTPClient.makeWith(mock: stack)
 
 		// Test.
+		let restServiceProvider = RestServiceProviderStub(
+			results: [
+				.success(SubmissionTANModel(submissionTAN: "fake")),
+				.success(SubmissionTANModel(submissionTAN: "fake"))
+			]
+		)
+		let resource = RegistrationTokenResource(
+			sendModel: SendRegistrationTokenModel(
+				token: "dummyRegToken"
+			)
+		)
+		restServiceProvider.load(resource) { _ in
+			expectation.fulfill()
+		}
+		let fakeRequestResource = RegistrationTokenResource(
+			isFake: true,
+			sendModel: SendRegistrationTokenModel(
+				token: "dummyRegToken"
+			)
+		)
 
-		client.getTANForExposureSubmit(forDevice: "dummyRegToken") { _ in expectation.fulfill() }
-		client.getTANForExposureSubmit(forDevice: "dummyRegToken", isFake: true) { _ in expectation.fulfill() }
-		client.getTestResult(forDevice: "dummyDevice") { _ in expectation.fulfill() }
-		client.getTestResult(forDevice: "dummyDevice", isFake: true) { _ in expectation.fulfill() }
+		restServiceProvider.load(fakeRequestResource) { _ in
+			expectation.fulfill()
+		}
+		
+		client.getTestResult(forDevice: "dummyDevice") { _ in
+			expectation.fulfill()
+		}
+		client.getTestResult(forDevice: "dummyDevice", isFake: true) { _
+			in expectation.fulfill()
+		}
```

Comment:
I think the whole test should be refactored. Instead of looping over different requests and compare the sizes, wouldn't it be easier to define a constant with expected size inside the test class and the split up test into separated tests for each request?

This will lead to a better separation of the old and new client and the test would be easier to understand and maintain.